### ISSUE TITLE
Source tarball fails to build if `git` tool is available

### ISSUE THREAD

**Issue by falkTX** [surge-synthesizer/surge#2491]
See title.

With the 1.7.1, trying to build on Linux fails right away.

```
cd obj-x86_64-linux-gnu && cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_RUNSTATEDIR=/run "-GUnix Makefiles" -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_AUTOGEN_VERBOSE=ON -DCMAKE_INSTALL_LIBDIR=lib/x86_64-linux-gnu ..
-- The CXX compiler identification is GNU 9.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Compiler Version is 9.3.0
CMake Warning at CMakeLists.txt:53 (message):
   Building VST2 using /usr/include - this is an unsupported configuration


-- Looking for std::filesystem::path::preferred_separator
-- Looking for std::filesystem::path::preferred_separator - found
-- Using std::filesystem
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'cairo'
--   Found cairo, version 1.16.0
-- Checking for module 'fontconfig'
--   Found fontconfig, version 2.13.1
-- Checking for module 'x11'
--   Found x11, version 1.6.9
-- Checking for module 'xcb'
--   Found xcb, version 1.14
-- Checking for module 'xcb-cursor'
--   Found xcb-cursor, version 0.1.1
-- Checking for module 'xcb-util'
--   Found xcb-util, version 0.4.0
-- Checking for module 'xcb-xkb'
--   Found xcb-xkb, version 1.14
-- Checking for module 'xkbcommon'
--   Found xkbcommon, version 0.10.0
-- Checking for module 'xkbcommon-x11'
--   Found xkbcommon-x11, version 0.10.0
-- Checking for module 'sndfile'
--   Found sndfile, version 1.0.28
-- Found LibSndFile: /usr/lib/x86_64-linux-gnu/libsndfile.so  
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_AUTOGEN_VERBOSE
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LOCALSTATEDIR
    CMAKE_INSTALL_RUNSTATEDIR
    CMAKE_INSTALL_SYSCONFDIR


-- Build files have been written to: /Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu
   dh_auto_build -O-Scmake
        cd obj-x86_64-linux-gnu && make -j1 "INSTALL=install --strip-program=true"
make[1]: Entering directory '/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu'
/usr/bin/cmake -S/Shared/Personal/FOSS/Repo/Debian-Plugins/surge -B/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu/CMakeFiles /Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[2]: Entering directory '/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu'
make -f CMakeFiles/git-info.dir/build.make CMakeFiles/git-info.dir/depend
make[3]: Entering directory '/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu'
cd /Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /Shared/Personal/FOSS/Repo/Debian-Plugins/surge /Shared/Personal/FOSS/Repo/Debian-Plugins/surge /Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu /Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu /Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu/CMakeFiles/git-info.dir/DependInfo.cmake --color=
Scanning dependencies of target git-info
make[3]: Leaving directory '/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu'
make -f CMakeFiles/git-info.dir/build.make CMakeFiles/git-info.dir/build
make[3]: Entering directory '/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu'
cd /Shared/Personal/FOSS/Repo/Debian-Plugins/surge && /usr/bin/cmake -D PROJECT_VERSION_MAJOR=1 -D PROJECT_VERSION_MINOR=7 -D SURGESRC=/Shared/Personal/FOSS/Repo/Debian-Plugins/surge -D SURGEBLD=/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu -D AZURE_PIPELINE= -P /Shared/Personal/FOSS/Repo/Debian-Plugins/surge/cmake/versiontools.cmake
-- Found Git: /usr/bin/git (found version "2.25.1") 
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
-- Setting up surge version
--   git hash is  and branch is 
--   buildhost is falkTX-PC.Home
--   buildarch is x86_64
-- Developer Local Build
CMake Error at cmake/versiontools.cmake:49 (if):
  if given arguments:

    "STREQUAL" "main"

  Unknown arguments specified


make[3]: *** [CMakeFiles/git-info.dir/build.make:60: CMakeFiles/git-info] Error 1
make[3]: Leaving directory '/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu'
make[2]: *** [CMakeFiles/Makefile2:289: CMakeFiles/git-info.dir/all] Error 2
make[2]: Leaving directory '/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu'
make[1]: *** [Makefile:87: all] Error 2
make[1]: Leaving directory '/Shared/Personal/FOSS/Repo/Debian-Plugins/surge/obj-x86_64-linux-gnu'
dh_auto_build: error: cd obj-x86_64-linux-gnu && make -j1 "INSTALL=install --strip-program=true" returned exit code 2
make: *** [debian/rules:27: binary] Error 2
```

To reproduce:
1. download surge 1.7.1 source tarball
2. make sure `git` CLI tool is available
3. try to build surge.

**Comment by baconpaul** [comment]
oh sure. yeah we embed the git hash in the binary so if git status doesn't work the build won't work.

**Comment by falkTX** [comment]
What is the best way to go about this then?
I can comment out the lines that try to get git version, but feels a bit wrong.
For tagged source tarball releases, it is best to read the information version from some static file. This can be auto-generated of course, a file that only exists for the release tar.

**Comment by baconpaul** [comment]
Yeah indeed!

I think the right thing is, indeed, to have a fallback which is generated at tarball time and modify cmake to read it.

For now, you can add a cmake that is something like 'if( NOT DEFINED ${GIT_BRANCH} ) set( GIT_BRANCH "kxrelease" )` 

the other way you could resolve this, of course, is not build from tarball but instead check out the `release_1.7.1` tag from our GitHub and build in that source tree.

(And of course, if we hadn't removed the .git dir from the tarball this would work, I just realized! Sigh).

But really you are the only person building from tarball as opposed from repo so we will definitely have a bit of work.

Thoughts?

**Comment by falkTX** [comment]
> But really you are the only person building from tarball as opposed from repo

For now..

Pretty much any other Linux distribution will want to package with a release tarball.
It is just that surge has not been packaged on these others yet. ArchLinux might be an exception, they might accept a git tag.

can cmake read files and put the content into a variable?
that seems the easiest path forward.

**Comment by baconpaul** [comment]
Yeah sure. The code which is breaking is all in 'cmake/versiontools.cmake` and you can see it is set up with a branch. Make that else() instead of an else() something like `if( ! gitfound || ! defined( git_branch ) )` for now and you should build.

**Comment by baconpaul** [comment]
I realized this issue is still open and I fixed it in 66b0cfaed90c241810569284c2983b4c1e079aca so I'm going to close it. The nightly has that asset as will 1.7.2.
